### PR TITLE
fix(history): add query_params echo to _fetch_statistics response

### PR DESCRIPTION
**Pull request by Patch76**
## Design decisions

**Why `statistic_types` echoes raw caller input, not the resolved default (`all_stat_types`)?**
`_fetch_statistics` resolves `None` to all six types internally via
`all_stat_types`. Echoing `all_stat_types` would reflect internal expansion,
not what the caller requested. Round-trip observability means: what did the
caller ask for? `statistic_types=None` is a valid and meaningful caller state.
This also matches how `_fetch_history` echoes `minimal_response` and
`significant_changes_only` — as-received, not post-processed.

**Why not include `period` in `query_params`?**
`period_type` is already present at the top level of the response. Including
it in `query_params` would be redundant. `_fetch_history` does not echo
`period` either — `query_params` covers caller-controlled filtering params,
not time-range metadata.

**Why not rename `time_range` → `period` for consistency with history?**
`period` in history is the time range. `period_type` in statistics is the
aggregation interval — conceptually distinct. Out of scope here.

**Known edge case: `statistic_types=[]`**
`[]` is falsy, so `all_stat_types` falls back to the full six-type default while
`query_params.statistic_types` would show `[]`. This is pre-existing behavior in
`_fetch_statistics`, not introduced here. Tracked separately in #979.

## Type of change

- [x] 🐛 Bug fix
- [ ] ✨ New feature
- [ ] 📚 Documentation
- [ ] 🔧 Maintenance/refactor
- [ ] 🧪 Tests only
- [ ] 💥 Breaking change

## Testing

- [x] `test_statistics_query_params_default` (unit, `TestStatisticsPagination`) — asserts `statistic_types=None`, `limit=_DEFAULT_HISTORY_LIMIT`, `offset=0` on default call
- [x] `test_statistics_query_params_roundtrip` (unit, `TestStatisticsPagination`) — asserts exact round-trip of `statistic_types=["mean"]`, `limit=10`, `offset=5`
- [x] Both tests call `ha_get_history` (public tool layer) via `HistoryTools` mock, not `_fetch_statistics` directly — matching `test_history_pagination.py` established pattern
- [x] Removed E2E `test_get_statistics_query_params_in_response` (used `sun.sun` which has no `state_class` → no LTS → false-green on stock fixture)
- [x] Test fix: `"year"` re-added to `valid_periods` iteration list in E2E test (was incorrectly dropped; production `valid_periods` at L583 already includes it)
- [x] Existing `test_get_history_query_params_in_response` unaffected
- [x] All automated tests pass

## Checklist

- [x] I have updated documentation if needed

### CODE REVIEW

File: src/ha_mcp/tools/tools_history.py [hunk 1]
```diff
@@ -581,6 +581,10 @@ async def _fetch_statistics(
             "end": end_dt.isoformat(),
         },
         "statistic_types": all_stat_types,
+        "query_params": {
+            "period": period,
+            "statistic_types": statistic_types,
+        },
```
[the remark this answers is not in the document]
Comment (Patch76) [reply src/ha_mcp/tools/tools_history.py:694]: Finding 1 (period redundancy): accepted. period_type at the top level carries the same value — including it in query_params contradicts the stated design principle (time_range excluded for the same reason). Removed in the follow-up commit.

Finding 2 (statistic_types raw vs. resolved): the inconsistency with _fetch_history is real, but I believe the difference is intentional. limit=None in _fetch_history triggers active server-side truncation, so echoing the effective_limit reflects what the server actually did. statistic_types=None means "no filter applied" — the server returns all types regardless. Echoing all_stat_types would hide whether the caller constrained the request or not, which is the observability point. The values differ in kind: one is a computed bound, the other is a filter state. I've left statistic_types as raw caller input for this reason.

One additional finding surfaced during review: statistic_types=[] (empty list) is falsy, so all_stat_types resolves to the full six-type default — while query_params.statistic_types would show []. This is pre-existing behavior in _fetch_statistics, not introduced by this PR. Noted in the PR description as a known edge case and tracked as a follow-up item.

**Review by kingpanther13 — CHANGES_REQUESTED** [review]
The stated symmetry goal with `_fetch_history` is only half-met, and the test that's supposed to guard the change will no-op in CI against a stock HA fixture. Blockers below.

## Critical

**1. Echo block is missing `limit` and `offset`** (`tools_history.py:690-692`)

`_fetch_history`'s `query_params` block at L528-533 echoes four caller params: `minimal_response`, `significant_changes_only`, `limit`, `offset`. The new statistics block echoes only `statistic_types`. But `_fetch_statistics` accepts `limit` and `offset` (added by #964, coerced into `effective_limit`/`effective_offset` at L552/L568, used for paging at L649) — they're caller-controllable and belong in the echo for the symmetry claim to hold. Without them this PR is doing half the fix.

**2. `sun.sun` doesn't emit long-term statistics** (`test_history.py:616`)

`sun.sun` is a string-state entity with no `state_class`, so HA's recorder doesn't generate LTS for it. The sibling `test_get_statistics_single_entity` at L348-387 goes through `ha_search_entities` specifically to find a numeric sensor because `sun.sun` won't work. On a stock HA test fixture, `inner.get("success")` will be falsy, the early-exit fires, and the real assertions never execute.

**3. Bare `return` is a false-green, not a skip** (`test_history.py:629-631`)

`logger.info(...); return` on the unavailable-data path reports as PASS. Other tests in this file (L387, L438, L481, L675, L679, L689) correctly use `pytest.skip(...)`. Combined with #2, the test silently greens without exercising the new code path — which is the test's entire reason for existing.

## Important

**4. Assertion checks key presence only, not values** (`test_history.py:634-636`)

`assert "statistic_types" in qp` doesn't verify the contract the PR body promises ("echo raw caller input"). A bug assigning the wrong value would pass. Should be:

```python
assert qp["statistic_types"] is None  # default call
```

Plus a second invocation with `statistic_types=["mean"]` asserting `qp["statistic_types"] == ["mean"]` to anchor the echo contract. (Different from the deferred `[]` case — asserting a non-empty list round-trips is in scope.)

**5. PR body internally inconsistent**

The example code block at the top of the description shows `"period": period` inside `query_params`, but the shipped diff omits `period` entirely and the test explicitly asserts `"period" not in qp`. Either the example is stale from an earlier draft or the decision to drop `period` wasn't back-propagated. Please update the PR body to match what's shipped.

**6. Unrelated scope creep — `"year"` dropped from periods list** (`test_history.py:442`)

Not mentioned in the PR body, not on this PR's topic. `"year"` isn't in `valid_periods` (`tools_history.py:583`: `5minute/hour/day/week/month`), so the removal is a correctness fix — but either document it in the PR body or split into its own commit.

## Suggestions (non-blocking)

- Move `test_get_statistics_query_params_in_response` into `TestGetHistoryStatisticsSource` class (alongside `test_get_statistics_single_entity`) for co-location with sibling statistics tests and inherit the class-level markers.

## Strengths

- Thoughtful design-decisions section in the PR body (once the `period` inconsistency is cleaned up)
- Correctly asserts `period` is absent — avoids duplication with `period_type`
- Small, focused production diff
- Good scope discipline elsewhere — `statistic_types=[]` is appropriately deferred to #979

Once #1, #2, #3 are addressed (and ideally #4 for a stronger contract guard), this is clean.

**Comment by Patch76** [comment]
Thanks for the thorough review — every point is well-founded. Here's what's in the follow-up commits.

## Critical items

**#1 — `limit` and `offset` missing from echo**

Fixed. The `query_params` block now includes all three caller-controllable parameters:

```python
"query_params": {
    "statistic_types": statistic_types,  # raw caller input
    "limit": effective_limit,             # coerced int
    "offset": effective_offset,           # coerced int
}
```

`period` is intentionally absent (same reasoning as before: redundant with top-level `period_type`; `_fetch_history` doesn't echo its period equivalent either).

**#2 — `sun.sun` has no `state_class` → test assertions never execute**

Removed the E2E test entirely. Unit coverage is the right layer for this contract.

**#3 — `logger.info(); return` is a false-green, not a skip**

Resolved by removing the E2E test (see #2). No `return`-on-unavailable path in the unit replacement.

## Important items

**#4 — Assertions check key presence only, not values**

Two unit tests added to `TestStatisticsPagination` in `test_history_pagination.py`, following the `_patch_ws` + `HistoryTools(mock_client).ha_get_history` pattern from that file:

- `test_statistics_query_params_default`: asserts `qp["statistic_types"] is None`, `qp["limit"] == 100`, `qp["offset"] == 0`
- `test_statistics_query_params_roundtrip`: asserts exact values when `statistic_types=["mean"]`, `limit=10`, `offset=5` — any wrong assignment fails

**#5 — PR body example shows `"period": period` but shipped diff omits it**

PR body updated. `period` removed from the example; `limit` and `offset` added. The design-decisions section is updated to explain why `period` is absent.

**#6 — `"year"` dropped from `valid_periods` in test, undocumented**

Fixed: `"year"` re-added to the iteration list in the E2E test. Production code at L583 already has it — the test drop was a bug. Documented in the commit message and PR body.

---

All 12 CI checks green.

**Review by kingpanther13 — APPROVED** [review]
32fc3ced addresses all four blockers cleanly.

- `query_params` block now echoes all three caller-controllable params (`statistic_types`, `limit`, `offset`) — symmetry with `_fetch_history` achieved
- Two new unit tests in `TestStatisticsPagination` exercise the dispatch path through `HistoryTools.ha_get_history(source="statistics", ...)` with exact-value assertions — stronger than my original ask
- `"year"` re-added to the periods iteration list (was a bug in the prior drop)
- PR body updated to match shipped code

Per AGENTS.md §Test Coverage Requirements, this falls squarely under the "minor parameter additions to well-tested tools" exemption — `ha_get_history` already has extensive E2E coverage and this PR's surface (a dict literal constructed from caller inputs) is exactly what unit tests can cover exhaustively. No E2E gap introduced on master either way.

One observation for future work, not a blocker: `query_params.statistic_types` echoes the raw caller input, so a caller passing `"mean,max"` (string) would see the string in `query_params` while the top-level `statistic_types` key has the parsed list. The unit tests only exercise the list path — worth noting if the "echo raw caller input" contract ever needs to be revisited, but consistent with the stated contract as shipped.

CI green, clean merge.
